src/main/java/org/hisp/dhis/cache/EntitiesCache.java

```diff
@@ -72,10 +79,59 @@ public void loadTeiTypeCache()
         }
     }
 
+    public void loadTeiCache()
+    {
+        this.teis = new ConcurrentHashMap<>();
+        for ( Program program : this.programs )
+        {
+            List<String> orgUnits = program.getOrgUnits();
+            orgUnits.parallelStream().forEach( ou -> {
+                // -- fetch a list of teis for each program + ou
+                List<Map> payload = getPayload(
+                    "/api/trackedEntityInstances?ou=" + ou + "&program=" + program.getUid() ).jsonPath()
+                        .getList( "trackedEntityInstances" );
+                int count = 0;
+                List<Tei> teisFromProgram = new ArrayList<>();
+                for ( Map map : payload )
+                {
+                    if ( count <= 10 ) // -- only add 10 teis for each ou
+                    {
+                        teisFromProgram.add( new Tei( (String) map.get( "trackedEntityInstance" ), program.getUid() ) );
+                        count++;
+                    } else {
+                        break;
+                    }
+                }
+                List<Tei> teis = this.teis.get( program.getUid() );
+                if ( teis != null && teis.size() != 0 )
+                {
+                    teis.addAll( teisFromProgram );
+                    this.teis.replace( program.getUid(), teis );
+                }
+                else
+                {
+                    this.teis.put( program.getUid(), teisFromProgram );
+                }
+            } );
+        }
+        //int teiSize = teis.values().stream().map(List::size).mapToInt(Integer::intValue).sum();
+        System.out.println( "teis by program" );
+        for ( String s : teis.keySet() )
+        {
+            System.out.println( "program: " + s + " - teis: " + teis.get( s ).size() );
+        }
+
+    }
+
     public void loadAll()
     {
         this.loadTeiTypeCache();
+        System.out.println("Tracked Entity Types loaded in cache [" + this.teiTypes.size() + "]");
         this.loadProgramCache();
+        System.out.println("Programs loaded in cache [" + this.programs.size() + "]");
+        this.loadTeiCache();
+        System.out.println("Tracked Entity Instances loaded in cache [" + this.teis.size() + "]");
```

Comment:
This will print the number of programs, not TEIs, so might be confusing

Reply:
Ah yes, fixed